src/Symfony/Component/Translation/Translatable/MoneyTranslatable.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Translatable;
+
+use Money\Currencies\ISOCurrencies;
+use Money\Formatter\DecimalMoneyFormatter;
+use Money\Money;
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Wrapper around PHP NumberFormatter for money
+ * The provided currency is used instead of the locale's currency.
+ *
+ * Implementation of the ICU recommendation to first format advanced parameters before translation.
+ *
+ * @see https://unicode-org.github.io/icu/userguide/format_parse/messages/#format-the-parameters-separately-recommended
+ *
+ * @author Sylvain Fabre <syl.fabre@gmail.com>
+ */
+class MoneyTranslatable implements TranslatableInterface
+{
+    private $value;
+    private $currency;
+    private $style;
+
+    private $formatters = [];
+
+    public function __construct($value, string $currency, int $style = \NumberFormatter::CURRENCY)
+    {
+        $this->value = $value;
+        $this->currency = $currency;
+        $this->style = $style;
+    }
+
+    public function trans(TranslatorInterface $translator, string $locale = null): string
+    {
+        if (!$locale) {
+            $locale = $translator->getLocale();
+        }
+
+        if (!isset($this->formatters[$locale])) {
+            $this->formatters[$locale] = new \NumberFormatter($locale, $this->style);
+        }
+
+        return $this->formatters[$locale]->formatCurrency($this->value, $this->currency);
+    }
+
+    /**
+     * Short-hand to instantiate from a Money instance.
+     */
+    public static function fromMoney(Money $money, int $style = \NumberFormatter::CURRENCY): self
+    {
+        $currencies = new ISOCurrencies();
+        $moneyFormatter = new DecimalMoneyFormatter($currencies, $style);
+
+        return new self($moneyFormatter->format($money), $money->getCurrency()->getCode());
```

Comment:
This is broken. `NumberFormatter` expects a number (`int|float`) but DecimalMoneyFormatter returns a formatted string already.

Reply:
Well spotted, I'll remove it as this is actually stupid: there is no point of using a `Money` object to turn it into float before formatting 🤦